contracts/FHE.sol

```diff
@@ -2443,239 +2417,256 @@ library FHE {
+    function getDecryptResult(euint64 input1) internal view returns (uint64) {
+        return uint64(Impl.getDecryptResult(euint64.unwrap(input1)));
+    }
+
+    /// @notice Gets the decrypted value from a previously decrypted euint128 ciphertext
+    /// @dev This function assumes the ciphertext has already been decrypted. Use getDecryptResultSafe for a safer alternative
+    /// @param input1 The euint128 ciphertext to get the decrypted value from
+    /// @return The decrypted uint128 value
+    function getDecryptResult(euint128 input1) internal view returns (uint128) {
+        return uint128(Impl.getDecryptResult(euint128.unwrap(input1)));
+    }
+
+    /// @notice Gets the decrypted value from a previously decrypted euint256 ciphertext
+    /// @dev This function assumes the ciphertext has already been decrypted. Use getDecryptResultSafe for a safer alternative
+    /// @param input1 The euint256 ciphertext to get the decrypted value from
+    /// @return The decrypted uint256 value
+    function getDecryptResult(euint256 input1) internal view returns (uint256) {
+        return uint256(Impl.getDecryptResult(euint256.unwrap(input1)));
+    }
+
+    /// @notice Gets the decrypted value from a previously decrypted eaddress ciphertext
+    /// @dev This function assumes the ciphertext has already been decrypted. Use getDecryptResultSafe for a safer alternative
+    /// @param input1 The eaddress ciphertext to get the decrypted value from
+    /// @return The decrypted address value
+    function getDecryptResult(eaddress input1) internal view returns (address) {
+        return address(uint160(Impl.getDecryptResult(eaddress.unwrap(input1))));
+    }
+
+    /// @notice Gets the decrypted value from a previously decrypted raw ciphertext
+    /// @dev This function assumes the ciphertext has already been decrypted. Use getDecryptResultSafe for a safer alternative
+    /// @param input1 The raw ciphertext to get the decrypted value from
+    /// @return The decrypted uint256 value
+    function getDecryptResult(uint256 input1) internal view returns (uint256) {
+        return Impl.getDecryptResult(input1);
+    }
+
+    /// @notice Safely gets the decrypted value from an ebool ciphertext
+    /// @dev Returns both the decrypted value and a flag indicating if decryption was successful
+    /// @param input1 The ebool ciphertext to get the decrypted value from
+    /// @return result The decrypted boolean value
+    /// @return decrypted Flag indicating if the value was successfully decrypted
+    function getDecryptResultSafe(ebool input1) internal view returns (bool result, bool decrypted) {
+        (uint256 _result, bool _decrypted) = Impl.getDecryptResultSafe(ebool.unwrap(input1));
+        return (_result != 0, _decrypted);
+    }
+
+    /// @notice Safely gets the decrypted value from a euint8 ciphertext
+    /// @dev Returns both the decrypted value and a flag indicating if decryption was successful
```

Comment:
/// @dev Returns the decrypted value and a flag indicating whether the decryption has finished